### PR TITLE
feat: migrate clean command to Git::Commands::Clean

### PR DESCRIPTION
**Pull request by jcouball**
This PR migrates the `clean` command as part of Phase 2 of the architectural redesign for v5.0.0.

## Changes

- **Create `Git::Commands::Clean`**: New command class in `lib/git/commands/clean.rb` using the Arguments DSL
  - Supports `force`, `force_force`, `d`, and `x` flags
  - Includes conflict validation between `:force` and `:force_force` options
  - Full YARD documentation with examples

- **Add RSpec tests**: Comprehensive test coverage in `spec/git/commands/clean_spec.rb`
  - 8 test cases covering all options and edge cases
  - Mocked execution context for fast, isolated unit tests

- **Update `Git::Lib#clean`**: Delegate to new command class
  - Maintains backward compatibility with legacy `:ff` option (converts to `:force_force`)
  - Uses keyword argument forwarding with `**opts`

- **Update implementation plan**: Mark `clean` as migrated (9/~50 commands complete)

## Verification

All tests pass:
- ✅ RSpec: 380 examples, 0 failures  
- ✅ TestUnit: 529 tests, 0 failures
- ✅ Rubocop: no offenses
- ✅ YARD: no documentation errors

## Migration Progress

This is command #9 in Phase 2 of the architectural redesign. See [3_architecture_implementation.md](redesign/3_architecture_implementation.md) for the full migration plan.

**Next task**: Migrate the `branch` command → `Git::Commands::Branch`

### CODE REVIEW

File: lib/git/lib.rb [hunk 1]
```diff
@@ -1157,16 +1158,11 @@ def reset(commit = nil, opts = {})
...
-      { keys: [:d],     flag: '-d',      type: :boolean },
-      { keys: [:x],     flag: '-x',      type: :boolean }
-    ].freeze
-
     def clean(opts = {})
-      args = build_args(opts, CLEAN_OPTION_MAP)
-      command('clean', *args)
+      # Convert legacy :ff option to :force_force for the new command class
+      opts = opts.dup
+      opts[:force_force] = opts.delete(:ff) if opts.key?(:ff)
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/lib.rb]: Added the deprecation and the test for the deprecation.

File: spec/git/commands/clean_spec.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    end
+
+    context 'with an unexpected positional argument' do
+      it 'raises an ArgumentError' do
+        expect { command.call('unexpected') }.to(
+          raise_error(ArgumentError, /Unexpected positional arguments: unexpected/)
+        )
+      end
+    end
+  end
```
[the remark this answers is not in the document]
Comment (jcouball) [reply spec/git/commands/clean_spec.rb:89]: Added the test as requested.

File: lib/git/commands/clean.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      #
+      # @param execution_context [Git::ExecutionContext, Git::Lib] the context for executing git commands
+      #
+      def initialize(execution_context)
+        @execution_context = execution_context
+      end
+
+      # Execute the git clean command
+      #
+      # @overload call(force: nil, d: nil, x: nil)
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/commands/clean.rb]: Updated.

File: spec/git/commands/clean_spec.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      it 'adds -ff to the command line' do
+        expect(execution_context).to receive(:command).with('clean', '-ff')
+        command.call(force_force: true)
+      end
+
+      it 'raises an ArgumentError when the :force argument is also given' do
+        expect { command.call(force: true, force_force: true) }.to(
+          raise_error(ArgumentError, /cannot specify :force and :force_force/)
+        )
+      end
```
[the remark this answers is not in the document]
Comment (jcouball) [reply spec/git/commands/clean_spec.rb]: Added the tests.
